### PR TITLE
kernel/vm: Make some functions static and remove dead code

### PR DESCRIPTION
**Pull request by astalke**
## Description

- MISRA
- Make `_page_map` and `_page_initSizes` static
- Remove dead function `_page_showSizes`

Fixes: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1071
Fixes: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1072
Fixes: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1073
Fixes: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1080
Fixes: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1082

JIRA: RTOS-836

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: ia32-generic-qemu

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by niewim19 — DISMISSED** [review]
- MISRA changes looks good, no functionality seems lost.
 - changes to functions satisfy issues
   -  https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1071
   -  https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1073
    - https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1072

Looks good to me.

File: vm/page.c [hunk 1]
```diff
@@ -120,28 +124,34 @@ void _page_free(page_t *p)
...
+	if ((p->addr & ((1 << (idx + 1)) - 1)) != 0) {
 		lh = p - (1 << idx) / SIZE_PAGE;
-	else
+	}
+	else {
 		rh = p + (1 << idx) / SIZE_PAGE;
+	}
 
-	while (lh >= pages.pages && (rh < pages.pages + (pages.allocsz + pages.freesz) / SIZE_PAGE) && (lh->flags & PAGE_FREE) && (rh->flags & PAGE_FREE) && (lh->idx == rh->idx) && (lh->addr + (1 << lh->idx) == rh->addr) && (idx < SIZE_VM_SIZES)) {
+	while ((lh >= pages.pages) && (rh < pages.pages + (pages.allocsz + pages.freesz) / SIZE_PAGE) && (lh->flags & PAGE_FREE) && ((rh->flags & PAGE_FREE) != 0) && (lh->idx == rh->idx) && (lh->addr + (1 << lh->idx) == rh->addr) && (idx < SIZE_VM_SIZES)) {
```
Comment (agkaminski) [inline vm/page.c:134]:
```suggestion
	while ((lh >= pages.pages) && (rh < pages.pages + (pages.allocsz + pages.freesz) / SIZE_PAGE) && ((lh->flags & PAGE_FREE) != 0) && ((rh->flags & PAGE_FREE) != 0) && (lh->idx == rh->idx) && ((lh->addr + (1 << lh->idx)) == rh->addr) && (idx < SIZE_VM_SIZES)) {
```

File: vm/page.c [hunk 2]
```diff
@@ -230,27 +244,13 @@ void _page_initSizes(void)
...
-	return;
-}
-
-
 static unsigned int page_digits(unsigned int n, unsigned int base)
 {
 	unsigned int d = 1;
 
-	while (n /= base)
+	while (n /= base) {
```
Comment (agkaminski) [inline vm/page.c:251]:
```suggestion
	while ((n /= base) != 0) {
```

File: vm/page.c [hunk 3]
```diff
@@ -103,11 +106,12 @@ void _page_free(page_t *p)
 	page_t *lh = p, *rh = p;
 
 #if 1
```
Comment (niewim19) [inline vm/page.c:108]: Not technically a dead code, but is this necessary?

Comment (agkaminski) [reply vm/page.c:108]: imho may be removed, it's not like anyone remembers why is this way it is

File: vm/page.c (hunk 1 above)
Comment (agkaminski) [reply vm/page.c]: I know I'm supper nitpicky, but I think you missed:
is:
`(lh->addr + (1 << lh->idx) == rh->addr)`
should be:
`((lh->addr + (1 << lh->idx)) == rh->addr)`

**Review by agkaminski — APPROVED** [review]
LGTM

**Review by agkaminski — DISMISSED** [review]

**Comment by niewim19** [comment]
@damianloew is this PR`s CI able to be repaired by rerunning campaign? Or do we hold it as error case?

**Comment by damianloew** [comment]
> @damianloew is this PR`s CI able to be repaired by rerunning campaign? Or do we hold it as error case?

Re-run should help, I've already done it :)

**Review by niewim19 — APPROVED** [review]
Thanks for implementing these chore changes!

File: vm/page.c [hunk 4]
```diff
@@ -196,14 +190,6 @@ page_t *_page_get(addr_t addr)
 }
 
 
-void vm_pageFreeAt(pmap_t *pmap, void *vaddr)
```
Comment (Darchiv) [inline vm/page.c:189]: Why is `vm_pageFreeAt()` removed only from here and not from vm/page-nommu.c or vm/page.h?

File: test/vm.c [hunk 5]
```diff
@@ -50,7 +50,7 @@ void test_vm_alloc(void)
 			break;
 		}
 
-		_page_free(p);
+		vm_pageFree(p);
```
Comment (Darchiv) [inline test/vm.c:53]: This technically changes code logic, as we use synchronized version of `pageFree` now. But it is just a test, so I'm not sure whether we should mark this change as bugfix/new feature or leave as chore.

File: vm/page.c (hunk 4 above)
Comment (astalke) [reply vm/page.c:189]: Because I didn't notice that I didn't add these files to a commit. I've fixed that now.

File: test/vm.c (hunk 5 above)
Comment (Darchiv) [reply test/vm.c:53]: phoenix-rtos/phoenix-rtos-project/issues/1080 is fixed there, so maybe mark the change as bugfix.

**Review by Darchiv — APPROVED** [review]

**Pull request merged**